### PR TITLE
Feat(server+client): preview account complete signup

### PR DESCRIPTION
**Pull request by abrl91**
Close: https://github.com/amplication/private-issues/issues/117

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-server/src/core/auth/auth.service.ts [hunk 1]
```diff
@@ -316,6 +316,57 @@ export class AuthService {
...
+        throw Error("Failed to create new Auth0 user");
+    }
+
+    const userEmail = existingAuth0User
+      ? currentAccount.previewAccountEmail
+      : auth0User.data.email;
+
+    const resetPassword = await this.resetAuth0UserPassword(userEmail);
+
+    if (!resetPassword.data)
```
Comment (morhag90) [inline packages/amplication-server/src/core/auth/auth.service.ts:341]: Could this object return undefined results?

File: packages/amplication-server/src/core/auth/auth.resolver.ts [hunk 2]
```diff
@@ -54,6 +54,14 @@ export class AuthResolver {
         });
   }
 
+  @Mutation(() => String)
+  @UseGuards(GqlAuthGuard)
+  async completeSignupWithBusinessEmail(
```
Comment (overbit) [inline packages/amplication-server/src/core/auth/auth.resolver.ts:59]: is this related to the PR?

Comment (abrl91) [reply packages/amplication-server/src/core/auth/auth.resolver.ts:59]: of course..it's the resolver

Comment (abrl91) [reply packages/amplication-server/src/core/auth/auth.resolver.ts:59]: I don't know why it seems like I did it. It's what Shimi did. Maybe something with the merges...

File: packages/amplication-server/src/core/auth/auth.service.ts (hunk 1 above)
Comment (abrl91) [reply packages/amplication-server/src/core/auth/auth.service.ts:341]: The data will be undefined, so it's ok...

File: packages/amplication-server/src/core/auth/auth.resolver.ts (hunk 2 above)
Comment (abrl91) [reply packages/amplication-server/src/core/auth/auth.resolver.ts:59]: yes...it's the resolver of my new endpoing

Comment (overbit) [reply packages/amplication-server/src/core/auth/auth.resolver.ts:59]: Then the naming is misleading as it might assume that is any business email and not preview only

**Review by morhag90 — APPROVED** [review]
